distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso

```diff
@@ -545,6 +546,89 @@ type Decimal
+       "round towards positive infinity." If use_bankers=True, then it uses
+       "round-half-even", also known as "banker's rounding".
+
+       Arguments:
+       - decimal_places: The number of decimal places to round to. Can be
+         negative, which results in rounding to positive integer powers of 10.
+         Must be between -15 and 15 (inclusive).
+       - use_bankers: Use banker's rounding
+
+       ! Error Conditions
+
+         If `decimal_places` is outside the range -15..15 (inclusive), an
+         `Illegal_Argument` error is thrown.
+
+       ? Negative decimal place counts
+         Rounding to `n` digits can be thought of as "rounding to the nearest
+         multiple of 10^(-n)". For negative decimal counts, this results in
+         rounding to the nearest positive integer power of 10.
+
+       ? Banker's rounding
+         Banker's rounding rounds half-way values to the nearest even number.
+
+       ? Infinities and NaN
+         - NaN.round == 0
+         - +Inf.round == +Inf
+         - -Inf.round == -Inf
+
+       > Example
+         Round to the nearest integer.
+
+            3.3 . round
+            # ==> 3
+
+       > Example
+         Round to two decimal places.
+
+            3.1415 . round 2
+            # ==> 3.14
+
+       > Example
+         Round to the nearest hundred.
+
+            1234.0 . round -2
+            # ==> 1200
+
+       > Example
+         Use Banker's Rounding.
+
+            2.5 . round use_bankers=True
+            # ==> 2
+    round : Integer -> Decimal ! Illegal_Argument
+    round self decimal_places=0 use_bankers=False =
+        case decimal_places < round_min_decimal_places || decimal_places > round_max_decimal_places of
+            True ->
+                msg = "round: decimal_places must be between " + round_min_decimal_places.to_text + " and " + round_max_decimal_places.to_text + "(inclusive), but was " + decimal_places.to_text
+                Error.throw (Illegal_Argument.Error msg)
+            False -> if self.is_nan || self.is_infinite then self else
+                case use_bankers of
+                    False ->
+                        scale = 10 ^ decimal_places
+                        ((self * scale) + 0.5).floor / scale
+                    True ->
+                        ## If the largest integer <= self is odd, use normal
+                           round-towards-positive-infinity rounding; otherwise,
+                           use round-towards-negative-infinity rounding.
+                        scale = 10 ^ decimal_places
+                        scaled_self = self * scale
+                        self_scaled_floor = scaled_self.floor
+                        is_even = (self_scaled_floor % 2) == 0
+                        case is_even of
+                            False ->
+                                (scaled_self + 0.5).floor / scale
+                            True ->
+                                (scaled_self - 0.5).ceil / scale
```

Comment:
```suggestion
                scale = 10 ^ decimal_places
                if use_bankers.not then ((self * scale) + 0.5).floor / scale else
                    ## If the largest integer <= self is odd, use normal
                        round-towards-positive-infinity rounding; otherwise,
                        use round-towards-negative-infinity rounding.
                    scaled_self = self * scale
                    self_scaled_floor = scaled_self.floor
                    is_even = (self_scaled_floor % 2) == 0
                    case is_even of
                        False ->
                            (scaled_self + 0.5).floor / scale
                        True ->
                            (scaled_self - 0.5).ceil / scale
```

Reply:
That's slightly more compact but I think harder to read.